core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -84,30 +143,157 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
       this.isTranslationCoordinator = userInfo.isTranslationCoordinator();
 
       if (this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER];
+        this.CONTRIBUTION_TYPES.push(
+          this.TAB_NAME_TRANSLATION_SUBMITTER,
+          this.TAB_NAME_TRANSLATION_REVIEWER);
       }
       if (this.isQuestionCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
-      }
-      if (this.isQuestionCoordinator && this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER,
+        this.CONTRIBUTION_TYPES.push(
           this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
+          this.TAB_NAME_QUESTION_REVIEWER);
       }
 
+      this.contributorDashboardAdminStatsBackendApiService
+        .fetchAssignedLanguageIds(username).then(
+          response => {
+            this.languageChoices = this.languageChoices.filter((
+                languageItem) =>
+              response.includes(languageItem.id));
+            this.selectedLanguage.language = (
+              this.languageChoices[0].language);
+            this.selectedLanguage.id = this.languageChoices[0].id;
+            this.translationReviewersCount = (
+              this.translationReviewersCountByLanguage[
+                this.selectedLanguage.id]);
+          });
+
+      this.filter = new ContributorAdminDashboardFilter(
+        [],
+        this.selectedLanguage.id,
+        null,
+        this.selectedLastActivity);
+
       this.updateSelectedContributionType(this.CONTRIBUTION_TYPES[0]);
+
+      this.loadingMessage = '';
       this.changeDetectorRef.detectChanges();
     });
 
+    this.lastActivity = [0, 7, 30, 90];
+
     this.contributorDashboardAdminStatsBackendApiService
-      .fetchCommunityStats().then(
-        (response) => {
-          this.translationReviewersCount = response.translation_reviewers_count;
-          this.questionReviewersCount = response.question_reviewers_count;
-        });
+      .fetchTopicChoices().then(
+        response => {
+          this.topics = response;
+          response.map((
+              topic) =>
+            this.allTopicNames.push(topic.topic));
+          this.applyTopicFilter();
+        }
+      );
+  }
+
+  toggleLanguageDropdown(): void {
+    this.languageDropdownShown = !this.languageDropdownShown;
+  }
+
+  toggleActivityDropdown(): void {
+    this.activityDropdownShown = !this.activityDropdownShown;
+  }
+
+  applyTopicFilter(): void {
+    this.selectedTopics.ids = this.selectedTopics.topics.map(
+      selectedTopic => {
+        const matchingTopic = this.topics.find(
+          topicChoice => topicChoice.topic === selectedTopic);
+        return matchingTopic ? matchingTopic.id : '';
```

Comment:
If this does occur, then we should probably tell the user with an error message toast, otherwise they will be confused as to why their topic filter is not working as intended.

Reply:
This won't occur in frontend ever cause the selectedTopics is getting selected from this.topics only.

Reply:
Yeah. That sounds reasonable.

Reply:
You mean an empty string? Yes, but the user selected a topic name in the filter, but we would pass an empty string for that selected topic name, so the filter would do nothing for that selected topic name. That seems like an error.

Reply:
Yeah that sounds reasonable. Added a condition to throw an error when there is a selected topic but doesn't match any from the list. PTAL.